DirectProgramming/DPC++FPGA/ReferenceDesigns/merge_sort/src/CMakeLists.txt

```diff
@@ -0,0 +1,108 @@
+# Profiling can be enabled when running cmake by adding the flag -DPROFILE_HW=1
+# e.g. cmake .. -DPROFILE_HW=1
+if(PROFILE_HW)
+    set(PROFILE_FLAG "-Xsprofile")
+endif()
+
+# Select the number of merge units to use. Must be a power of 2.
+# e.g. cmake .. -DNUM_MERGE_UNITS=16
+if(MERGE_UNITS)
+    set(MERGE_UNITS_FLAG "-DMERGE_UNITS=${MERGE_UNITS}")
+    message(STATUS "Number of merge units explicitly set to ${MERGE_UNITS}")
+endif()
+
+# Choose the random seed for the hardware compile
+# e.g. cmake .. -DSEED=7
+if(NOT DEFINED SEED)
+    # the default seed
+    set(SEED 3)
+else()
+    message(STATUS "Seed explicitly set to ${SEED}")
+endif()
+
+# A DPC++ ahead-of-time (AoT) compile processes the device code in two stages.
+# 1. The "compile" stage compiles the device code to an intermediate representation (SPIR-V).
+# 2. The "link" stage invokes the compiler's FPGA backend before linking.
+#    For this reason, FPGA backend flags must be passed as link flags in CMake.
+set(EMULATOR_COMPILE_FLAGS "-Wall ${WIN_FLAG} -fintelfpga ${ENABLE_USM} ${MERGE_UNITS_FLAG} -DFPGA_EMULATOR")
+set(EMULATOR_LINK_FLAGS "-fintelfpga ${ENABLE_USM} ${MERGE_UNITS_FLAG}")
+set(HARDWARE_COMPILE_FLAGS "-Wall ${WIN_FLAG} -fintelfpga ${ENABLE_USM} ${MERGE_UNITS_FLAG}")
+set(HARDWARE_LINK_FLAGS "-fintelfpga -Xshardware ${PROFILE_FLAG} -Xsparallel=2 -Xsseed=${SEED} -Xsboard=${FPGA_BOARD} ${ENABLE_USM} ${MERGE_UNITS_FLAG} ${USER_HARDWARE_FLAGS}")
+# use cmake -D USER_HARDWARE_FLAGS=<flags> to set extra flags for FPGA backend compilation
+
+###############################################################################
+### FPGA Emulator
+###############################################################################
+# To compile in a single command:
+#    dpcpp -fintelfpga -DFPGA_EMULATOR merge_sort.cpp -o merge_sort.fpga_emu
+# CMake executes:
+#    [compile] dpcpp -fintelfpga -DFPGA_EMULATOR -o merge_sort.cpp.o -c merge_sort.cpp
+#    [link]    dpcpp -fintelfpga merge_sort.cpp.o -o merge_sort.fpga_emu
+add_executable(${EMULATOR_TARGET} ${SOURCE_FILE})
+set_target_properties(${EMULATOR_TARGET} PROPERTIES COMPILE_FLAGS "${EMULATOR_COMPILE_FLAGS}")
+set_target_properties(${EMULATOR_TARGET} PROPERTIES LINK_FLAGS "${EMULATOR_LINK_FLAGS}")
+add_custom_target(fpga_emu DEPENDS ${EMULATOR_TARGET})
+
+###############################################################################
+### Generate Report
+###############################################################################
+# To compile manually:
+#   dpcpp -fintelfpga -Xshardware -Xsboard=<FPGA_BOARD> -fsycl-link=early merge_sort.cpp -o merge_sort_report.a
+set(FPGA_EARLY_IMAGE ${TARGET_NAME}_report.a)
+add_custom_target(report DEPENDS ${FPGA_EARLY_IMAGE})
+set(CMAKE_CXX_FLAGS_LIST ${CMAKE_CXX_FLAGS})
+separate_arguments(CMAKE_CXX_FLAGS_LIST)
+set(HARDWARE_LINK_FLAGS_LIST ${HARDWARE_LINK_FLAGS})
+separate_arguments(HARDWARE_LINK_FLAGS_LIST)
+add_custom_command(OUTPUT ${FPGA_EARLY_IMAGE}
```

Comment:
Please update your CMake to match the newest standard template.  I think the Generate Report section is the only change from what you have, but please double check.

Reply:
I copied the one from the CRR design in the tip of GitHub. I will base it off MVDR.

Reply:
Base it off of this: https://github.com/oneapi-src/oneAPI-samples/pull/492 (except for changing /EHsc to -EHsc - don't do that, as mentioned elsewhere)